ssh-u2f.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python
+#
+# -*- coding: utf-8 -*-
+#
+# 2016-03-03 Brandon Smith <freedom@reardencode.com>
+#            Initial Creation
+#
+# (c) Brandon Smith
+# Info: http://www.privacyidea.org
+#
+# This code is free software; you can redistribute it and/or
+# modify it under the terms of the GNU AFFERO GENERAL PUBLIC LICENSE
+# License as published by the Free Software Foundation; either
+# version 3 of the License, or any later version.
+#
+# This code is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU AFFERO GENERAL PUBLIC LICENSE for more details.
+#
+# You should have received a copy of the GNU Affero General Public
+# License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+__doc__ = """This is an ssh (and ssh-like) wrapper that uses pexpect to
+interact with privacyIDEA's pam_python module for u2f challenge/response.
+
+Usage:
+    Make executable
+    Symlink ssh-u2f, scp-u2f, sftp-u2f, mosh-u2f, etc. into your PATH
+    Call just like ssh, eg. "ssh-u2f name@example.com"
+"""
+from __future__ import print_function
+
+import getpass,os,re,signal,subprocess,sys
+import pexpect
+
+ssh = None
+
+def handler(signum, frame):
+    global ssh
+    if ssh:
+        ssh.kill(signum)
+    sys.exit(signum)
+
+signal.signal(signal.SIGQUIT, handler)
+signal.signal(signal.SIGTERM, handler)
+signal.signal(signal.SIGINT, handler)
+
+def winch_handler(signum, frame):
+    global ssh
+    if ssh:
+        rows, cols = os.popen('stty size', 'r').read().split()
+        ssh.setwinsize(int(rows), int(cols))
+
+signal.signal(signal.SIGWINCH, winch_handler)
+
+try:
+    command = os.path.splitext(os.path.basename(__file__))[0].split("-")[0]
+except:
+    command = None
+
+ssh = pexpect.spawn(command or "ssh", sys.argv[1:])
+winch_handler(None, None)
+
+def passthrough():
+    print()
+    sys.stdout.write(ssh.match.group())
+    try:
+        ssh.interact()
+    except UnboundLocalError:
+        # Work around bug in pexpect 3.1
+        pass
+    sys.exit(0)
+
+index = ssh.expect(["Authenticated with partial success.",
+                    "[^ \r\n]+",
+                    pexpect.EOF])
+
+if index == 0:
```

Comment:
Ah, are you using privacyIDEA PAM as your primary authentication as opposed to secondary?  I totally forgot about that use case (we require publickey + U2F)